DirSearcherClient/Program.cs

```diff
@@ -97,27 +91,30 @@ static void Search(string searchterm, string tenant)
                     Console.WriteLine(ee.Message);
                 }
 
-                if (jResult["odata.error"] != null || jResult["value"] == null)
+                if (jResult == null || jResult["odata.error"] != null || jResult["value"] == null)
                 {
                     Console.ForegroundColor = ConsoleColor.Red;
                     Console.WriteLine("Error on search");
                 }
                 Console.ForegroundColor = ConsoleColor.Green;
-                if (jResult.Count == 0)
+                if (jResult != null)
```

Comment:
>if (jResult != null) [](start = 16, length = 20)

invert and return or return above where you anyway state we are in an error situation.

Reply:
I returned above. Thanks for catching this.